src/BenchmarkDotNet.Core/Engines/GcStats.cs

```diff
@@ -79,22 +82,34 @@ public static GcStats FromForced(int forcedFullGarbageCollections)
 
         private static long GetAllocatedBytes(bool isDiagnosticsEnabled)
         {
-#if NETCOREAPP11 // when MS releases new version of .NET Runtime to nuget.org
-            return GC.GetAllocatedBytesForCurrentThread(); // https://github.com/dotnet/corefx/pull/12489
-#elif CLASSIC
-            if (!isDiagnosticsEnabled || RuntimeInformation.IsMono()) // Monitoring is not available in Mono, see http://stackoverflow.com/questions/40234948/how-to-get-the-number-of-allocated-bytes-
+            if (!isDiagnosticsEnabled 
+                || RuntimeInformation.IsMono()) // Monitoring is not available in Mono, see http://stackoverflow.com/questions/40234948/how-to-get-the-number-of-allocated-bytes-
                 return 0;
 
             // "This instance Int64 property returns the number of bytes that have been allocated by a specific 
             // AppDomain. The number is accurate as of the last garbage collection." - CLR via C#
             // so we enforce GC.Collect here just to make sure we get accurate results
             GC.Collect();
+#if CORE
+            return getAllocatedBytesForCurrentThread.Invoke();
+#elif CLASSIC
+
             return AppDomain.CurrentDomain.MonitoringTotalAllocatedMemorySize;
-#else
-            return 0; // currently for .NET Core
 #endif
         }
 
+        private static Func<long> GetAllocatedBytesForCurrentThread()
+        {
+            // for some versions of .NET Core this method is internal, 
+            // for some public and for others public and exposed ;)
```

Comment:
I have been waiting for a long time for such a thing to happen. That's healthy IMHO :D